examples/complete/stacks/catalog/terraform/test-component-override-2.yaml

```diff
@@ -14,6 +14,12 @@ components:
       # By using `component` attribute, `test-component-override` inherits from `test-component` and can override its variables
       # In this example, variables for each service are overridden in `catalog/services/service-?-override.*`
       component: "test/test-component-override"
+      # Setting `deployable: false` explicitly prohibits the component from being deployed
+      # and Spacelift stack from being created for the component (even if `settings.spacelift.workspace_enabled: true`)
+      # `terraform apply` and `terraform deploy` will fail with an error
+      # All other terraform commands on this component will succeed
+      # If `deployable` attribute is not specified, it defaults to `true`
+      deployable: true
```

Comment:
What I don't like about `deployable` is that we have a subcommand for `deploy`, and it sounds like this relates to that. But it also affects `plan` and `apply`. So is there another way? Perhaps we should define the type of component, and call it `type: base` or something like `public` and `private` in object-oriented programming.

Reply:
`type: base` does not solve the issue.
We have base components and derived components (no virtual).
Both base and derived are deployable.
This PR solves 2 issues:

1. Prevent deployment of a base component that is not supposed to be deployed (it's just a base with default vars/settings pointing to the TF component
2. Allow setting `settings.spacelift.workspace_enabled` globally and not setting it on all derived components explicitly

Let's say we have this config

```
# global
settings:
  spacelift:
    workspace_enabled: true

components:
  terraform:
    aurora-postgres:
      settings:
        spacelift:
          workspace_enabled: false

    aurora-postgres-example:
      component: aurora-postgres

```

We don't want to create a Spacelift stack for `aurora-postgres` since it's not `deployable` (it's just a base with defaults for many derived components).

The above config will not work (the global `workspace_enabled: true` will not be used) since `aurora-postgres-example` is derived from `aurora-postgres` and inherits everything from it including `workspace_enabled: false`.
The Spacelift stack for `aurora-postgres-example` will not be created.

To solve this, we can use this in EVERY derived component:

```
    aurora-postgres-example:
      component: aurora-postgres
      settings:
        spacelift:
          workspace_enabled: true
```

which is not DRY and does not take the global `workspace_enabled: true` into account.


-----

Using `deployable` like this:

```
# global
settings:
  spacelift:
    workspace_enabled: true

components:
  terraform:
    aurora-postgres:
     deployable: false
      settings:
        spacelift:
          workspace_enabled: false

    aurora-postgres-example:
      component: aurora-postgres

```

serves a few purposes:

1. `aurora-postgres` is not supposed to be deployed and will not be provisioned accidentally (via CLI) - the CLI will throw an error 
2. Regardless of what value the users put into `workspace_enabled` (`true` or `false`), a Spacelift stack for `aurora-postgres` will not be created (it's not supposed to).

This will work with `workspace_enabled: false` and `workspace_enabled: true` (if somebody forgot to disable Spacelift stack for the derived component - happens all the time). 


```
components:
  terraform:
    aurora-postgres:
     deployable: false
      settings:
        spacelift:
          workspace_enabled: false

components:
  terraform:
    aurora-postgres:
     deployable: false
      settings:
        spacelift:
          workspace_enabled: true
```

And it will allow the derived component `aurora-postgres-example` to always work in Spacelift w/o explicitly adding `workspace_enabled: true` to it since it will be inherited from the global setting.